plugins/slo/pkg/apis/slo/slo.proto

```diff
@@ -121,65 +107,36 @@ message Alert {
 }
 
 message ServiceLevelObjective {
-  string id = 1;
-  string name = 2;
-  string datasource = 3;
-  string description = 4;
-  repeated Service services = 5;
-  string monitorWindow = 6; // 7d, 28d or 30d
-  string metricDescription = 8;
-  string budgetingInterval = 9; // in the format 5m, 1h etc..
-  repeated Label labels = 10;
-  repeated Target targets = 11;
-  repeated Alert alerts = 12;
-}
-
-message Target {
-  uint64 ValueX100 = 1;
-  google.protobuf.Duration timeWindow = 2; // FIXME: should be changed to str
-}
-
-message ServiceLevelObjectiveList {
-  repeated ServiceLevelObjective items = 1;
-}
-
-// TODO : rework this
-message SLOStatus {
-  core.Reference slo = 1;
-  int64 timestamp = 2;
-  Evaluation status = 3;
-  Evaluation budget = 4;
+  string name = 1;
+  string datasource = 2;
+  string monitorWindow = 3; // 7d, 28d or 30d
+  string budgetingInterval = 4; // in the format 5m, 1h etc..
+  Target target = 5;
+  repeated Label labels = 6;
+  repeated Alert alerts = 7;
 }
 
-message MetricList {
-  repeated Metric items = 1;
+message CreateSLORequest {
+  ServiceLevelObjective SLO = 1;
+  repeated Service services = 2;
 }
 
-
-message SetStateRequest {
-  core.Reference slo = 1;
-  State state = 2;
+message Services {
+  repeated Service items = 1;
 }
 
-// Contains all the SLO data like, error budget, burn rate, etc...
-message State {
-  bytes data = 1;
+message CreatedSLOs {
+  repeated core.Reference items = 1;
 }
 
-/// Human readable pre-configured metrics
-message Formula {
+message SLOImplData {
   string id = 1;
-  string name = 2;
-  string description = 3;
+  ServiceLevelObjective SLO = 2;
+  Service service = 3;
 }
 
-message FormulaList {
-  repeated Formula items = 1;
-}
-
-message Evaluation {
-  double rate = 1;
-  double period = 2;
+message Target {
+  uint64 ValueX100 = 1;
```

Comment:
rename with lowercase v (for consistent json key naming)

Reply:
Done